modules/native_streaming_client_module/src/native_device_impl.cpp

```diff
@@ -120,10 +143,16 @@ void NativeDeviceHelper::setupProtocolClients(const ContextPtr& context)
     };
     configProtocolClient = std::make_unique<ConfigProtocolClient<NativeDeviceImpl>>(context, sendRequestCallback, nullptr);
 
-    auto receiveConfigPacketCb =
-        [this](const PacketBuffer& packet)
+    ProcessConfigProtocolPacketCb receiveConfigPacketCb =
+        [this](PacketBuffer&& packetBuffer)
     {
-        return this->receiveConfigPacket(packet);
+        auto packetBufferPtr = std::make_shared<PacketBuffer>(std::move(packetBuffer));
+        processingIOContext.dispatch(processingStrand.wrap(
+            [this, packetBufferPtr]()
+            {
+                this->processConfigPacket(std::move(*packetBufferPtr));
```

Comment:
Could it ever happen that NativeDeviceHelper is destroyed and the dispatch callback is still executing? If so, "this" could be invalid here?

Edit: probably not because thread is started and stopped inside NativeDeviceHelper